src/main/java/com/mobilegenomics/f5n/controller/DataController.java

```diff
@@ -281,8 +281,12 @@ private static void calculateJobFailureRate() {
 
     // Calculate New Job Arrival rate and update total Idle Jobs
     private static void calculateNewJobArrivalRate() {
-        newJobArrivalRate = (idleWrapperObjectList.size() - totalIdleJobs) / (float) statWatchTimerInMinutes;
-        totalIdleJobs = idleWrapperObjectList.size();
+        if (filePrefixes.size() - totalIdleJobs <= 0) {
```

Comment:
isn't filePrefixes list is an ever growing list, meaning that the files inside the folder will not be deleted. if this is true, it is not necessary to have if else conditions.

Reply:
> isn't filePrefixes list is an ever growing list, meaning that the files inside the folder will not be deleted. if this is true, it is not necessary to have if else conditions.

Yes, ever growing for a single server session. But, once the server is stopped(by button) it'll be cleared.  Thus this would be better, i think